packages/blocks/src/__internal__/rich-text/rich-text-operations.ts

```diff
@@ -20,8 +21,17 @@ import {
   focusPreviousBlock,
   focusBlockByModel,
   supportsChildren,
+  getPageTitle,
 } from '../utils/index.js';
 
+function focusTextEndAppendText(input: HTMLTextAreaElement, text = '') {
```

Comment:
In this case, you are directly mutating the DOM. But when it comes to non-native undo/redo, the single source of truth is always the model state (see https://github.com/toeverything/blocksuite/blob/master/packages/blocks/src/page-block/default/default-page-block.ts#L293-L298). So instead, calling `page.updateBlock(model, { title: newTitle })` and waiting for the state update should be the best practice.